CRM/Emailfiling/Hook/BuildForm/AddIsOriginalEmlAttachedField.php

```diff
@@ -46,13 +46,17 @@ public function run($formName, CRM_Core_Form &$form) {
    *
    * @param string $formName
    *   Form name.
+   * @param CRM_Core_Form $form
+   *   Form Class object.
    *
    * @return bool
    *   True if hook should run, false otherwise.
    */
-  private function shouldRun($formName) {
-    if ($formName === 'CRM_Admin_Form_MailSettings') {
-      return TRUE;
+  private function shouldRun($formName, CRM_Core_Form $form) {
+    if ($formName === 'CRM_Admin_Form_MailSettings' && $form->getVar('_id')) {
+      if (($_GET['action'] ?? '') !== 'delete') {
```

Comment:
can we do something like $form->getAction() !== CRM_Core_Action::DELETE and  can it be made part of first if so we could have only one if